linera-execution/src/runtime.rs

```diff
@@ -503,15 +538,21 @@ impl SyncRuntimeInternal<UserServiceInstance> {
 }
 
 impl<UserInstance> SyncRuntime<UserInstance> {
-    fn new(runtime: SyncRuntimeInternal<UserInstance>) -> Self {
-        SyncRuntime(Arc::new(Mutex::new(runtime)))
+    fn into_inner(mut self) -> Option<SyncRuntimeInternal<UserInstance>> {
+        let handle = self.0.take().expect(
+            "`SyncRuntime` should not be used after its `inner` contents have been moved out",
+        );
+        let runtime = Arc::into_inner(handle.0)?.into_inner().expect(
+            "`SyncRuntime::into_inner` should only be called \
+            after loaded applications have been cleared",
```

Comment:
Isn't this the `expect` that fails if someone else locked the mutex and then panicked? What does that have to do with the loaded applications?

Reply:
Oops, thanks for catching this! I fixed the message.